lightning/src/ln/channelmanager.rs

```diff
@@ -1591,15 +1605,48 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref> ChannelMan
 									htlc_id: prev_htlc_id,
 									incoming_packet_shared_secret: incoming_shared_secret,
 								};
-								channel_state.claimable_htlcs.entry(payment_hash).or_insert(Vec::new()).push(ClaimableHTLC {
+
+								let mut total_value = 0;
+								let htlcs = channel_state.claimable_htlcs.entry((payment_hash, if let &Some(ref data) = &payment_data {
```

Comment:
4d5666b

I think it was intentional but we miss a check here, if `payment_secret` doesn't match  expected value for that `payment_hash`, specs requires to to fail the HTLC. Sadly, doing so is likely to create an oracle to guess `payment_secret`, a processing node may withhold a HTLC shard and iterate with probing same-hash-HTLC until cracking `payment_secret`. If it's the penultimate node and your only link, it may even doing so without any time-bound.

Deviating from the spec, and withholding a likely-malicious HTLC shard means that's easy to DoS channel by once seeing locked a same-hash-HTLC until MPP timing expiration, a malicious node may hit `max_accepted_htlcs`, and so likely to prevent MPP settlement.

Hmmmm?

Reply:
I don't see how? payment_secret is a random 32 byte value. You can guess it until the end of the universe and still come up empty.